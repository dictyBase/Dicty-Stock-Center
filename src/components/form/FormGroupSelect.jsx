import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Flex, Box } from 'rebass'
import { ControlLabel, FormGroup, FormControlSelect, HelpBlock } from 'styles'

export default class FormGroupSelect extends Component {
  displayName = 'Select field component'

  static propTypes = {
      field: PropTypes.object.isRequired,
      list: PropTypes.array.isRequired,
      defaultValue: PropTypes.string,
      defaultTitle: PropTypes.string
  }

  static defaultProps = {
      defaultValue: 'select',
      defaultTitle: '-- Select --'
  }

  render() {
      const { field, list, defaultValue, defaultTitle } = this.props
      const hasError = field.touched && field.error
      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ 1 / 4 } ml={ 2 } mb={ 2 }>
            <ControlLabel>{ this.props.children }</ControlLabel>
          </Box>
          <Box w={ 2 / 3 } mr={ 2 }>
            <FormControlSelect {...field} value={ field.value || '' }>
              <option value={ defaultValue }>{ defaultTitle }</option>
              { list &&
                list.map(item => {
                    return (
                    <option key={ list.indexOf(item) } value={ item }>
                      { item }
                    </option>
                  )
                }) }
            </FormControlSelect>
            { hasError && <HelpBlock>{ field.error }</HelpBlock> }
          </Box>
        </Flex>
      </FormGroup>
    )
  }
}
