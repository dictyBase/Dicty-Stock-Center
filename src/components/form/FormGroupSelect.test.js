import React from 'react'
import renderer from 'react-test-renderer'
import FormGroupSelect from './FormGroupSelect'

test('matching a snapshot of FormGroupSelect', () => {
    const field = {
        touched: true
    }
    const list = [
        'USA',
        'Iceland',
        'Japan'
    ]

    const component = renderer.create(
    <FormGroupSelect field={ field } list={ list } />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
