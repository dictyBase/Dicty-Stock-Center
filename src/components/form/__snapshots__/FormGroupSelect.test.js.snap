// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of FormGroupSelect 1`] = `
<div
  className="sc-VJcYb eoqXel"
>
  <div
    className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
      is={null}
    >
      <label
        className="sc-fONwsr jwUYlu"
      />
    </div>
    <div
      className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
      is={null}
    >
      <select
        className="sc-cqCuEk bJVHks"
        value=""
      >
        <option
          value="select"
        >
          -- Select --
        </option>
        <option
          value="USA"
        >
          USA
        </option>
        <option
          value="Iceland"
        >
          Iceland
        </option>
        <option
          value="Japan"
        >
          Japan
        </option>
      </select>
    </div>
  </div>
</div>
`;
